rasa/nlu/utils/bilou_utils.py

```diff
@@ -240,22 +244,75 @@ def _add_bilou_tags_to_entities(
                 bilou[end_token_idx] = f"L-{label}"
 
 
-def _get_entity_positions(entities: List[Tuple[int, int, Text]]) -> Set[int]:
-    entity_positions = set()
+def ensure_consistent_bilou_tagging(predicted_tags: List[Text]) -> List[Text]:
+    """
+    Ensure predicted tags follow the BILOU tagging schema.
 
-    for start_pos, end_pos, label in entities:
-        for i in range(start_pos, end_pos):
-            entity_positions.add(i)
+    We assume that starting B- tags are correct. Followed tags that belong to start
```

Comment:
this should also be represented in confidences then, like we should remove the confidence for tags that were substituted. Should it rather be majority vote, like if all the tags except B-tag is the same?

Reply:
If we have an entity that covers multiple words, we take the lowest confidence score among those words. We could leave out the "misclassified" tag in this calculation.
I did not want to update the logic for now. Was thinking of using majority vote, but we might want to change this in a separate PR.

Reply:
in this calculation the confidence is basically the confidence of the first entity